pyquil/device.py

```diff
@@ -0,0 +1,315 @@
+##############################################################################
+# Copyright 2016-2018 Rigetti Computing
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+##############################################################################
+from collections import namedtuple
+
+import numpy as np
+from typing import Union, Sequence
+
+Qubit = namedtuple("Qubit", ["id", "type", "dead"])
```

Comment:
I personally don't like namedtuples, they don't have great IDE support and they are going to replaced by dataclasses once we're all on Python 3. Plus you can't docstring the members. Just write a short class with init method yourself if that's all you're using it for.